SWIPs/swip-message_to_honey.md

```diff
@@ -26,42 +25,75 @@ This SWIP is part of a series of SWIPs (but can be implemented on it's own). To
+* Querying the oracle will return a `messagePrices` object, specifying a `Time to Live (TTL)` in seconds and `prices` object objects. The `prices` object contains an entry for each messageType (`swarmMessage`) and `swarmMessage` maps a `validFrom` to a respective `price`. Taken together, the oracle returns: 
+```json
+{
+  "messagePrices": 
   {
-    TTL, 
-    prices: [0: {validFrom, msgPrices[]}, n:{...}]
+    "TTL": <TTLValue>, 
+    "prices": {
+      "swarmMessageX": {
+        "validFrom0": <price0>, 
+        ...,                
+        "validFromN": <priceN>
+      }
+      ...,
+      "swarmMessageY": {
+        "validFrom0": <price0>,
+        ...,
+        "validFromN": <priceN>
+      }
+    }
   }
 }
 ```
-* After `TTL` expires, nodes will query the price oracle for a new `honeyPrices` object. 
-* The `TTL` is set as a state-variable in the smart-contract and can be updated.
-* To ensure that it is not possible that a new entry in the `prices` array becomes valid before all nodes are updated, the smart-contract or update-policy must ensure that the `validFrom` of new prices must be at least `TTL` seconds in the future.
-* Upon start-up, nodes will look at the contents of their local cache and query the price oracle if no `honeyPrices` object is found, or the `TTL` of their cached `honeyPrices` object has expired. 
-* If the oracle cannot be reached when the `TTL` of the last `honeyPrices` object is expired, nodes will continue to apply the `prices` as instructed by the expired `honeyObject` and will start to query the oracle at a more regular interval to re-establish connection. 
+An example of this could be:
+```json
+{
+   "messagePrices": {
+      "TTL": 3600,
+      "chunkRequest": {
+        "1566469257": 1000,             
+        "1566472857": 1010
+      },
+      "chunkDelivery": {
+        "1566469257": 500,
+        "1566476457": 400
+      }
+   }
+}
+```
+* Any answer from the oracle will be valid for the node for `TTL` seconds.
+* The applied price is chosen by looking up the price corresponding to the validFrom which is in the most recent past.
+* After the `messagePrices` object expires, nodes will query the price oracle for a new `messagePrices` object. 
+* The source-code should contain a fallback value for messagePrices which is updated every release to reflect the most up-to-date message prices at that time.
+* Upon start-up, nodes will look at the contents of their local cache. If no `messagePrices` object is found, or the `TTL` of their cached `messagePrices` object has expired, the msgOracle will be queried. If the oracle can't be reached the fallback value will be used.
+* The `TTL` is set as a variable inside the implementation of the oracle and can be updated.
+* To ensure that a new entry in the 'smarmMessage' object can't become valid before all nodes are updated, the smart-contract or update-policy must ensure that the `validFrom` of new prices must be at least `TTL` seconds in the future.
```

Comment:
there should be no hyphen in `smart contract` nor in `update pocliy`

Reply:
Changed